src/fidesops/models/policy.py

```diff
@@ -115,6 +150,42 @@ class Policy(Base):
         backref="policies",
     )  # Which client created the Policy
 
+    @classmethod
+    def create_or_update(cls, db: Session, *, data: Dict[str, Any]) -> FidesopsBase:
+        """Overrides base create or update to add custom error for drp action already exists"""
+        db_obj = None
+        if data.get("id") is not None:
+            # If `id` has been included in `data`, preference that
+            db_obj = cls.get(db=db, id=data["id"])
+        elif data.get("key") is not None:
+            # Otherwise, try with `key`
+            db_obj = cls.get_by(db=db, field="key", value=data["key"])
```

Comment:
It's better to start this override by calling the inherited `create_or_update` method (https://github.com/ethyca/fidesops/blob/main/src/fidesops/db/base_class.py) to retrieve the DB class.

Reply:
wait, confused what you mean by call the base `create_or_update`, since that actually creates/updates in the db, and I need to add pre-logic to validate DRP action

Reply:
When you override `create_or_update` on `FidesopsUser`, you can still override `FidesopsBase.create_or_update` e.g.

```
class FidesopsUser:
    ...

        @classmethod
    def create_or_update(cls, db: Session, *, data: Dict[str, Any]) -> FidesopsBase:
        """Validate this object's data before deferring to the superclass on update"""
        instance = super().create(db=db, data=data)
        # mutate instance further
        return instance
```

Reply:
So I won't actually call `super().create_or_update()` from `Policy` but rather `FidesopsBase.get_by_key_or_id()`?